### ISSUE TITLE
[Enhancement]: Structural use of the notices wrapper

### ISSUE THREAD

**Issue by Elmer789** [woocommerce/woocommerce#31485]
### Describe the solution you'd like

It seems the woocommerce_output_all_notices function should wrap all notices inside the ‘woocommerce-notices-wrapper’ class. But in the end it doesn’t. So it’s really hard to change the appearance of notices as they are not all inside the same wrapper.

I have made a (partial) overview:

Shop page
Type: error, Content: Insufficient Stock, Wrapper used: Yes

Cart page
Type: success message, Content: Cart updated, Wrapper used: Yes
Type: error, Content: Coupon not valid, Wrapper used: Yes
Type: ?, Content: Empty cart, Wrapper used: NO
Type: success message, Content: Product removed from cart, Wrapper used: Yes
Type: success message, Content: Coupon applied, Wrapper used: Yes

Checkout page
Type: info, Content: Have a coupon?, Wrapper used: NO
Type: error, Content: Required fields missing, Wrapper used: NO
Type: error, Content: Coupon not valid, Wrapper used: NO
Type: success message, Content: Coupon applied, Wrapper used: NO

On top of that, the structure is not consistent. Based on the code I analysed I expect the structure to be (expressed as classes):
woocommerce/woocommerce-notices-wrapper/woocommerce-error (or -info or -message)

Sometimes the wrapper is just left out and sometimes the wrapper is replaced by another subclass (for example woocommerce-form-coupon-toggle or woocommerce-NoticeGroup woocommerce-NoticeGroup-checkout).

I’m using WooCommerce 5.9.0 and OceanWP theme (child). I have tested the behaviour mentioned above with Twenty Twenty-One theme and no plugins activated (except for WC of course)

There is probably a good reason for this mess, but could you please reconsider restoring the basic structure?

This would make it much easier to apply style sheets (and JS) to the notices.

**Comment by voxoff79** [comment]
yes this is very annoying on cart the woocommerce-error added via code wc_add_notice() its not wrapped in a  woocommerce-notices-wrapper but the woocommerce-error in you put an invalid coupon code for example is in a woocommerce-notices-wrapper

**Comment by lkraav** [comment]
:100: 

It's even worse with payment method box and other container structures used across Checkout and various My Account screens.

**Comment by pierorocca** [comment]
Hi @Elmer789 - thanks for the report.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

This appears to still be relevant, so I've marked it ready for the engineering team to pick up.

What I'm seeing: The inconsistency reported in 2021 still exists on trunk. WooCommerce has a canonical `woocommerce_output_all_notices()` helper that wraps notices in `.woocommerce-notices-wrapper`, but classic checkout coupon notices are rendered inside `.woocommerce-form-coupon-toggle` without that wrapper, checkout JavaScript prepends `.woocommerce-NoticeGroup-*` containers directly, empty-cart output uses `.wc-empty-cart-message`, and auth templates call `wc_print_notices()` directly. This is a valid enhancement request for more consistent notice markup. Implementation should be handled carefully because several of these wrappers are targeted by existing scripts/themes.

**Comment by pierorocca** [comment]
Claude Code notes:

## The Problem

WooCommerce has a `woocommerce_output_all_notices()` function that is *supposed* to wrap all notices in a consistent `<div class="woocommerce-notices-wrapper">` outer element. But in practice, the wrapper is only added sometimes — depending on which page you're on and which notice type is rendered.

**The notice templates themselves** (`notices/error.php`, `notices/success.php`, `notices/notice.php`) only output the inner element (`<ul class="woocommerce-error">`, `<div class="woocommerce-message">`, etc.) with no wrapper at all. The wrapper is applied by the caller — but inconsistently:

- **Cart page** — error and success notices get the wrapper; the "empty cart" info message does not
- **Checkout page** — almost nothing gets the wrapper; instead, different container classes are used (`woocommerce-NoticeGroup-checkout`, `woocommerce-form-coupon-toggle`)
- **Shop page** — the wrapper is present

`★ Insight ─────────────────────────────────────`
- **The inconsistency is architectural**: the wrapper responsibility is split between templates (which don't add it) and the calling context (which adds it only sometimes). A cleaner design would have the templates always emit a self-contained structure.
- **The real-world pain** is for theme/plugin developers: you can't write a single CSS selector like `.woocommerce-notices-wrapper .woocommerce-error { ... }` that reliably targets all error notices across all pages. You have to know which page you're on and target different class combinations.
- **Checkout is the worst offender** because it uses completely different class names (`woocommerce-NoticeGroup-checkout`) instead of the standard `woocommerce-error`, breaking even the inner-element selectors.
`─────────────────────────────────────────────────`

The fix would be making `woocommerce_output_all_notices()` (or the individual notice templates) consistently emit the wrapper on every page, every time.

<img width="1932" height="1666" alt="Image" src="https://github.com/user-attachments/assets/aba1a47f-23f4-495d-a354-8f3ab5c86462" />

<img width="940" height="223" alt="Image" src="https://github.com/user-attachments/assets/77542298-112c-4204-bb25-dea995efb835" />
